From 23fe372f02b96c9fe33d49c210361f78eff9c8ed Mon Sep 17 00:00:00 2001
From: "Thulasimani,Sivakumar" <sivakumar.thulasimani@intel.com>
Date: Wed, 23 Mar 2016 15:06:19 +0530
Subject: [PATCH 12/19] FOR_UPSTREAM [VPG]: drm/i915: Change sleep to delays
 for DPCD ops

DPCD operations are time sensitive and using sleep based calls such
as msleep results in some calls taking upto 10ms to be returning.
Hence this patch changes such calls in the dpcd path to
usleep_range. this will guarantee that the required delay is
satisfied and will also ensure that we continue normal
processing immediately after this delay.

Change-Id: I9eede5c10bc30c502271c6f801f529ab44c113e2
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-30099
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Reviewed-on: https://android.intel.com:443/498187
---
 drivers/gpu/drm/i915/intel_dp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 1235635..1228793 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -679,7 +679,7 @@ intel_dp_aux_ch(struct intel_dp *intel_dp,
 		status = I915_READ_NOTRACE(ch_ctl);
 		if ((status & DP_AUX_CH_CTL_SEND_BUSY) == 0)
 			break;
-		msleep(1);
+		usleep_range(1000, 1100);
 	}
 
 	if (try == 3) {
-- 
1.9.1

